### PR TITLE
feat: show welcome modal when wallet imported

### LINKED ISSUE #2754: wire up welcome to the new shapeshift app modal on imported wallet event
### Overview

When a user loads the new mobile app, if they have a mnemonic stored in the previous mobile app an event will fire that we can use to welcome the new user.

https://github.com/shapeshift/web/blob/347897af9cd93a37a31b7defc378de7cd47ebed3/src/plugins/mobile/index.tsx#L23

### References and additional details

Note - this is not the same event as a user importing a mnemonic from the membership endpoint. 

### Acceptance Criteria

# AC

When a user migrates from the old app to the new app and their wallet is carried over, we display them the intended modal and welcome message. 

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

On mobile when the wallet is imported successfully, we will show a welcome modal that will ask the user to backup their recovery phrase or allow them to skip it. This should only fire on the import of the wallet.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2754 

## Risk

By default this modal is not set to show, but once that event fires it will show. There is a risk that it blocks the user from entering their passphrase via modal stacking.

## Testing

Import mobile wallet, on success you should see this welcome modal.

### Engineering

We are using the store to dispatch a change to user preferences `setWelcomeModal` we set that to `true` and when they hit continue or dismiss we set it to false.
 
### Operations

Import mobile wallet, on success you should see this welcome modal.

Close and re-open the app, to confirm that you do not see this modal again.

### CODE REVIEW

File: src/state/slices/preferencesSlice/preferencesSlice.ts [hunk 1]
```diff
@@ -90,5 +92,8 @@ export const preferences = createSlice({
     setCurrencyFormat(state, { payload }: { payload: { currencyFormat: CurrencyFormats } }) {
       state.currencyFormat = payload.currencyFormat
     },
+    setWelcomeModal(state, { payload }: { payload: { show: boolean } }) {
```
Comment (gomesalexandre) [inline src/state/slices/preferencesSlice/preferencesSlice.ts:95]: Nitpick but `setShowWelcomeModal` would be more declarative and avoid calling `setWelcomeModal` with an object, but totally happy with this implementation

File: src/components/Modals/MobileWelcome/views/Notice.tsx [hunk 2]
```diff
@@ -17,6 +19,11 @@ export const Notice = () => {
   const handleRecoveryClick = () => {
     handleClose()
     open({ preventClose: true })
+    store.dispatch(preferences.actions.setWelcomeModal({ show: false }))
+  }
+  const handleDismissClick = () => {
```
Comment (gomesalexandre) [inline src/components/Modals/MobileWelcome/views/Notice.tsx:24]: `useCallback()`, I'm not so concerned if existing event handlers aren't tackled in PRs, but any event handlers or expressions passed down as props which are not memoized make it harder for us to get full memoization across the app (https://github.com/shapeshift/web/pull/2699) in

File: src/components/Modals/MobileWelcome/views/Notice.tsx [hunk 3]
```diff
@@ -17,6 +19,11 @@ export const Notice = () => {
   const handleRecoveryClick = () => {
```
Comment (gomesalexandre) [inline src/components/Modals/MobileWelcome/views/Notice.tsx:19]: Not required for this PR but as this was touched, can we `useCallback()` while at it?

File: src/components/Layout/Header/Header.tsx [hunk 4]
```diff
@@ -128,6 +131,7 @@ export const Header = () => {
               <Box display={{ base: 'none', md: 'block' }}>
                 <UserMenu />
               </Box>
+              <Button onClick={() => mobileWelcomeModal.open({})}>Fire Some shit</Button>
```
Comment (gomesalexandre) [inline src/components/Layout/Header/Header.tsx:134]: Revert file

File: src/App.tsx [hunk 5]
```diff
@@ -45,5 +50,11 @@ export const App = () => {
     }
   }, [shouldUpdate, toast, translate])
 
+  useEffect(() => {
+    if (showWelcomeModal) {
```
Comment (gomesalexandre) [inline src/App.tsx:54]: For extra safety
```suggestion
    if (showWelcomeModal && !isWelcomeModalOpen) {
```

File: src/App.tsx [hunk 6]
```diff
@@ -5,17 +5,22 @@ import { useToast } from '@chakra-ui/toast'
...
+import { selectShowWelcomeModal } from 'state/slices/selectors'
 
 export const App = () => {
   const shouldUpdate = useHasAppUpdated()
   const toast = useToast()
   const toastIdRef = useRef<ToastId | null>(null)
   const updateId = 'update-app'
   const translate = useTranslate()
+  const showWelcomeModal = useSelector(selectShowWelcomeModal)
+  const { mobileWelcomeModal } = useModal()
```
Comment (gomesalexandre) [inline src/App.tsx:23]:
```suggestion
  const { mobileWelcomeModal, isOpen: isWelcomeModalOpen } = useModal()
```

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Conceptual stamp as logic looks sane, added a few requested changes, once this is cleaned up let's get this in

**Review by gomesalexandre — DISMISSED** [review]

**Review by gomesalexandre — APPROVED** [review]
Re-stamp after the origin/develop merge and the removal of duplicated env config key following said merge

**Comment by cjthompson** [comment]
Mobile wallets don't have passwords so you won't get a password modal stacking.

@reallybeard were you able to test this? If not, I can.

**Comment by reallybeard** [comment]
> Mobile wallets don't have passwords so you won't get a password modal stacking.
> 
> @reallybeard were you able to test this? If not, I can.

@cjthompson I was only able to test if it got triggered manually. Would be great if you could test it in the app.

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**